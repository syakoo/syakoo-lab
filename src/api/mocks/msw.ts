import { handleAPIFetchArtFav } from "@/api/fav/arts/fetch-art-fav.mocks";
import { handleAPIIncrementArtFav } from "@/api/fav/arts/increment-art-fav.mocks";
import { handleAPISyncArtsFav } from "@/api/fav/arts/sync-arts-fav.mocks";

export const defaultHandlers = [
  handleAPIFetchArtFav,
  handleAPIIncrementArtFav,
  handleAPISyncArtsFav,
];
