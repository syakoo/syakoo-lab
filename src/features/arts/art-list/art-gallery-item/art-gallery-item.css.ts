import { style } from "@vanilla-extract/css";

import { theme } from "@/shared/design-system/theme.css";

export const artGalleryItemStyles = {
  image: style({
    aspectRatio: "1 / 1",
    objectFit: "cover",
    width: "100%",
    height: "auto",
    borderRadius: theme.radius[200],
  }),
  titleWrapper: style({
    display: "-webkit-box",
    overflow: "hidden",
    WebkitLineClamp: "1",
    WebkitBoxOrient: "vertical",
  }),
  heartButtonWrapper: style({
    width: 26,
  }),
};
