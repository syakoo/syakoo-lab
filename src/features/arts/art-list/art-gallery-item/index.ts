export { ArtGalleryItem } from "./art-gallery-item";
