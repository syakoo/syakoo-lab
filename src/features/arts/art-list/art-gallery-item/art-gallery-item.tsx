"use client";

import Image from "next/image";

import { api } from "@/api";
import { Col, FlexItem, Row } from "@/design-system/layout";
import { H3, Link } from "@/design-system/ui";
import type { ArtHead } from "@/features/arts/_models/types";
import { HeartButton } from "@/features/arts/_shared/heart-button";
import { artPaths } from "@/features/arts/config/paths";

import { artGalleryItemStyles } from "./art-gallery-item.css";

type ArtGalleryItemProps = {
  head: ArtHead;
};

export const ArtGalleryItem: React.FC<ArtGalleryItemProps> = ({ head }) => {
  const handleClickHeartButton = () =>
    void api.incrementArtFav({ id: head.id });

  return (
    <Col as="article" gap="50">
      <div>
        <Link display="block" href={artPaths.detail(head.id)}>
          <Image
            alt={head.title}
            className={artGalleryItemStyles.image}
            height={head.size.height}
            src={head.imgUrl}
            width={head.size.width}
          />
        </Link>
      </div>
      <Row align="center" justify="spaceBetween">
        <div className={artGalleryItemStyles.titleWrapper}>
          <H3 size="75">{head.title}</H3>
        </div>
        <FlexItem shrink={0}>
          <div className={artGalleryItemStyles.heartButtonWrapper}>
            <HeartButton onClick={handleClickHeartButton} />
          </div>
        </FlexItem>
      </Row>
    </Col>
  );
};
