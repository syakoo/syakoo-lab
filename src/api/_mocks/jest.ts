import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";

/**
 * jest 用のモックサーバー起動関数
 */
export function setupJestMockServer(...handlers: RequestHandler[]) {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
}
