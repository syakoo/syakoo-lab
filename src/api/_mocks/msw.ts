import { defaultHandlers as favArtsDefaultHandlers } from "../fav/arts/_mocks/msw";

export const defaultHandlers = [...favArtsDefaultHandlers];
