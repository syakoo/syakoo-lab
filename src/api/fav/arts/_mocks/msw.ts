import { type HttpHandler, http, delay, HttpResponse } from "msw";

import {
  FetchArtFavResponseBody,
  fetchArtFavPath,
} from "@/api/fav/arts/fetchArtFav";
import { incrementArtFavPath } from "@/api/fav/arts/incrementArtFav";
import { syncArtsFavPath } from "@/api/fav/arts/syncArtsFav";

export const handleSyncArtsFav: HttpHandler = http.post(
  syncArtsFavPath(),
  async () => {
    await delay(1000);
    return new HttpResponse();
  },
);

export const handleFetchArtFav: HttpHandler = http.get(
  fetchArtFavPath(":id"),
  async () => {
    await delay(1000);
    return HttpResponse.json<FetchArtFavResponseBody>({ fav: 10 });
  },
);

export const handleIncrementArtFav: HttpHandler = http.post(
  incrementArtFavPath(":id"),
  async () => {
    await delay(1000);
    return HttpResponse.json<FetchArtFavResponseBody>({ fav: 11 });
  },
);

export const defaultHandlers: HttpHandler[] = [
  handleSyncArtsFav,
  handleFetchArtFav,
  handleIncrementArtFav,
];
