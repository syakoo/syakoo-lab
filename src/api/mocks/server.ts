import { createMiddleware } from "@mswjs/http-middleware";
import cors from "cors";
import express from "express";

import { supabaseUrl } from "@/api/supabase/_shared/config";

import { defaultHandlers } from "./msw";

/**
 * ローカル環境用のモックサーバー起動関数
 */
export const runServer = () => {
  const app = express();

  app.use(cors());
  app.use(express.json());
  app.use(createMiddleware(...defaultHandlers));
  app.use((_req, res) => {
    res.status(404).json({
      error: "Mock not found",
    });
  });

  const { port } = new URL(supabaseUrl);
  app.listen(port);
};
