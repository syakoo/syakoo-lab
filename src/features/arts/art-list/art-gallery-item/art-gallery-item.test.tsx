import { setupJestMockServer } from "@/api/mocks/jest";
import { defaultHandlers } from "@/api/mocks/msw";
import { testStories } from "@/test/test-stories";

import * as stories from "./art-gallery-item.stories";

setupJestMockServer(...defaultHandlers);

describe("ArtGalleryItem", () => {
  testStories(stories);
});
