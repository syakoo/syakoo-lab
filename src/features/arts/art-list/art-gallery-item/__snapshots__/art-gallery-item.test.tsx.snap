// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtGalleryItem スナップショット LongText 1`] = `
<body>
  <div>
    <div
      style="max-width: 500px;"
    >
      <article
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_column__12xne02k flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <div>
          <a
            class="link_linkStyle__jc9bkl0 link_linkStyle_display_block__jc9bkl4"
            href="/arts/sample-id-2"
          >
            <img
              alt="Sample Art Title Sample Art Title Sample Art Title Sample Art Title Sample Art Title Sample Art Title"
              class="art-gallery-item_artGalleryItemStyles_image__1jhogya0"
              data-nimg="1"
              decoding="async"
              height="1500"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F3063D4%2Fffffff%2F2000x1500.png&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F3063D4%2Fffffff%2F2000x1500.png&w=3840&q=75 1x"
              style="color: transparent;"
              width="2500"
            />
          </a>
        </div>
        <div
          class="flex_flexStyle__12xne02i flex_flexStyle_align_center__12xne02s flex_flexStyle_variants_align_center__12xne027 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_spaceBetween__12xne02o flex_flexStyle_variants_justify_spaceBetween__12xne023"
        >
          <div
            class="art-gallery-item_artGalleryItemStyles_titleWrapper__1jhogya1"
          >
            <h3
              class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_bold__xbx8xut"
            >
              Sample Art Title Sample Art Title Sample Art Title Sample Art Title Sample Art Title Sample Art Title
            </h3>
          </div>
          <div
            class="flex_flexItemStyle__12xne0219 flex_flexItemStyle_shrink_0__12xne021a"
          >
            <div
              class="art-gallery-item_artGalleryItemStyles_heartButtonWrapper__1jhogya2"
            >
              <button
                aria-label="heart button"
                class="heart-button_heartButtonStyles_root__gy39e23"
                type="button"
              >
                <div
                  class="heart-button_heartButtonStyles_iconWrapper__gy39e24"
                >
                  <svg
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>
</body>
`;

exports[`ArtGalleryItem スナップショット Sample 1`] = `
<body>
  <div>
    <div
      style="max-width: 500px;"
    >
      <article
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_column__12xne02k flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <div>
          <a
            class="link_linkStyle__jc9bkl0 link_linkStyle_display_block__jc9bkl4"
            href="/arts/sample-id-1"
          >
            <img
              alt="Sample Art Title"
              class="art-gallery-item_artGalleryItemStyles_image__1jhogya0"
              data-nimg="1"
              decoding="async"
              height="1500"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F3063D4%2Fffffff%2F2000x1500.png&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F3063D4%2Fffffff%2F2000x1500.png&w=3840&q=75 1x"
              style="color: transparent;"
              width="2500"
            />
          </a>
        </div>
        <div
          class="flex_flexStyle__12xne02i flex_flexStyle_align_center__12xne02s flex_flexStyle_variants_align_center__12xne027 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_spaceBetween__12xne02o flex_flexStyle_variants_justify_spaceBetween__12xne023"
        >
          <div
            class="art-gallery-item_artGalleryItemStyles_titleWrapper__1jhogya1"
          >
            <h3
              class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_bold__xbx8xut"
            >
              Sample Art Title
            </h3>
          </div>
          <div
            class="flex_flexItemStyle__12xne0219 flex_flexItemStyle_shrink_0__12xne021a"
          >
            <div
              class="art-gallery-item_artGalleryItemStyles_heartButtonWrapper__1jhogya2"
            >
              <button
                aria-label="heart button"
                class="heart-button_heartButtonStyles_root__gy39e23"
                type="button"
              >
                <div
                  class="heart-button_heartButtonStyles_iconWrapper__gy39e24"
                >
                  <svg
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>
</body>
`;
